geppetto.js/geppetto-ui/src/tree-viewer/showcase/examples/TreeShowcaseSearch.js

```diff
@@ -0,0 +1,157 @@
+import React, { Component } from 'react';
+import Tree from '../../Tree';
+import '../TreeShowcase.less';
+
+
+export default class App extends Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      searchString: '',
+      searchFocusIndex: 0,
+      searchFoundCount: null,
+      treeData: this.getTreeData()
+    }
+  }
+  getTreeData () {
+    return [
+      {
+        title: 'San Diego, APRIL 25-26, 2009',
+        subtitle:
+            '1.5 T General Electric (GE) Signa Excite. 8-channel, transmit-receive head coil',
+        expanded: true,
+        children: [
+          {
+            title: '3-D T1-WEIGHTED FSPGR',
+            data: '3-D T1-WEIGHTED FSPGR',
+            active: true,
+          },
+          {
+            title: '3-D FAST SPIN ECHO',
+            data: '3-D FAST SPIN ECHO',
+          },
+          {
+            title: '2-D FAST SPIN ECHO',
+            data: '2-D FAST SPIN ECHO',
+          },
+          {
+            title: 'HIGH RES 3-D T1-WEIGHTED FSPGR',
+            data: 'HIGH RES 3-D T1-WEIGHTED FSPGR',
+          },
+        ],
+      },
+    ];
+  }
+
+  render () {
+    const { searchString, searchFocusIndex, searchFoundCount } = this.state;
+
+    // Case insensitive search of `node.title`
+    const customSearchMethod = ({ node, searchQuery }) =>
```

Comment:
keep the render function clear and move function declaration to class or global level.

Each time this component calls render() the functions will be recreated and get a new reference. Since these functions are passed to child components as props, the child components also re-render although it might not be necessary